Parser/asdl_c.py

```diff
@@ -1140,8 +1140,188 @@ def visitModule(self, mod):
     {NULL}
 };
 
+static PyObject *
+ast_repr_max_depth(AST_object *self, int depth);
+
+static PyObject *
+ast_repr_list(PyObject *list, int depth)
+{
+    struct ast_state *state = get_ast_state();
+    if (state == NULL) {
+        return NULL;
+    }
+
+    Py_ssize_t length = PySequence_Size(list);
+    if (length < 0) {
+        return NULL;
+    } else if (length == 0) {
+        return PyObject_Repr(list);
+    }
+
+    _PyUnicodeWriter writer;
+    bool is_list = PyList_Check(list);
+    PyObject *items[2];
```

Comment:
You'll probably make life easier if you initialize this to `{NULL, NULL}`, then you can `Py_XDECREF()` both items below without having to worry about whether they've been initialized.